YunoHost/yunohost
# Adding mDNS for .local domains

## The problem

`yunohost.local` domain broadcasting through mDNS is useful but a bit unflexible.

## Solution

- ~~Add a base `avahi-alias` service to publish as many domains as required~~
- ~~Add checks in `domain.py` to add these services whenever the user adds a domain ending by `.local`~~
- Use [Python-Zeroconf lib](https://github.com/jstasiak/python-zeroconf) to deploy a system service that broadcasts any .local domain over the network

## PR Status

- [x] Feature can be tested
- [x] Feature is integrated in diagnosis system
- [x] Seemingly no need for migrations
- [x] Feature is integrated in the regen_conf
- [x] Code complete and ready for review

## How to test

- Try using a `.local` domain during the postinstall
- Try adding a new `.local` domain
- Try adding `.local` subdomains: diagnosis should warn you it will not work. (BTW if someone can have a look to the Python library and figure out if/how CNAME records can be sent out...)
